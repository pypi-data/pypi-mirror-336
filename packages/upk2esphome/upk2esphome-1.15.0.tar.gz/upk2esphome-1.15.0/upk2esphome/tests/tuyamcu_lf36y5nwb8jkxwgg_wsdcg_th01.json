{
	"category": "wsdcg",
	"model": {
		"modelId": "000004ax7p",
		"services": [
			{
				"actions": [],
				"code": "",
				"description": "",
				"events": [],
				"name": "默认服务",
				"properties": [
					{
						"abilityId": 1,
						"accessMode": "ro",
						"code": "temp_current",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_c",
							"attribute": "1666",
							"trigger": "direct"
						},
						"name": "温度",
						"typeSpec": {
							"max": 600,
							"min": -200,
							"scale": 1,
							"step": 1,
							"type": "value",
							"typeDefaultValue": -200,
							"unit": "℃"
						}
					},
					{
						"abilityId": 2,
						"accessMode": "ro",
						"code": "humidity_value",
						"description": "",
						"extensions": {
							"iconName": "icon-shidu",
							"attribute": "1666",
							"trigger": "direct"
						},
						"name": "湿度",
						"typeSpec": {
							"max": 100,
							"min": 0,
							"scale": 0,
							"step": 1,
							"type": "value",
							"typeDefaultValue": 0,
							"unit": "%"
						}
					},
					{
						"abilityId": 3,
						"accessMode": "ro",
						"code": "battery_state",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_battery",
							"attribute": "1152",
							"trigger": "direct"
						},
						"name": "电池电量状态",
						"typeSpec": {
							"range": [
								"low",
								"middle",
								"high"
							],
							"type": "enum",
							"typeDefaultValue": "low"
						}
					},
					{
						"abilityId": 9,
						"accessMode": "rw",
						"code": "temp_unit_convert",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_mode",
							"attribute": "1152"
						},
						"name": "温标切换",
						"typeSpec": {
							"range": [
								"c",
								"f"
							],
							"type": "enum",
							"typeDefaultValue": "c"
						}
					}
				]
			}
		]
	}
}
