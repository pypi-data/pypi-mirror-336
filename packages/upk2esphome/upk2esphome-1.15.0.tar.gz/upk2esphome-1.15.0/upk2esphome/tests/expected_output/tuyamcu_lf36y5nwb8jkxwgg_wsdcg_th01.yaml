uart:
  rx_pin: RX1
  tx_pin: TX1
  baud_rate: REPLACEME

tuya:
  # DPIDs processed from schema model: 000004ax7p
  on_datapoint_update:
    - sensor_datapoint: 3
      datapoint_type: enum
      then:
        - text_sensor.template.publish:
            id: tuya_battery_state
            state: !lambda "return std::to_string(x);"

sensor:
  - platform: tuya
    sensor_datapoint: 1
    name: Temperature
    device_class: temperature
    unit_of_measurement: "\xB0C"
    accuracy_decimals: 1
    filters:
      - multiply: 0.1
  - platform: tuya
    sensor_datapoint: 2
    name: Humidity
    device_class: humidity
    unit_of_measurement: "%"

text_sensor:
  - platform: template
    id: tuya_battery_state
    name: Battery Level
    filters:
      - map:
          - 0 -> Low
          - 1 -> Medium
          - 2 -> High

select:
  - platform: tuya
    enum_datapoint: 9
    name: Unit Conversion
    entity_category: config
    optimistic: true
    options:
      0: C
      1: F
